core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -592,7 +592,13 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 int u = Integer.MAX_VALUE;
                 if (lowerTerm != null) {
                     l = parse(lowerTerm, true);
-                    if (includeLower == false || hasDecimalPart(lowerTerm)) {
+                    // if the lower bound is decimal:
+                    // - if the bound is positive then we increment it:
+                    //      if lowerTerm=1.5 then the (inclusive) bound becomes 2
+                    // - if the bound is negative then we leave it as is:
+                    //      if lowerTerm=-1.5 then the (inclusive) bound becomes -1 due to the call to longValue
+                    boolean lowerTermHasDecimalPart = hasDecimalPart(lowerTerm);
+                    if ((includeLower == false && !lowerTermHasDecimalPart) || (lowerTermHasDecimalPart && l > 0)) {
```

Comment:
I suspect this approach cannot work since -0.5 and +0.5 both parse to 0 when coerce is true, so you have no way to know whether the original value was positive or negative?